### ISSUE TITLE
Customizable fonts

### ISSUE THREAD

**Issue by BardiaB8** [hjiangsu/thunder#59]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to increase post and comment fonts. The default font is a little bit small for me.

**Describe the solution you'd like**
Add options in settings for "comment font" and "post font".

**Comment by hjiangsu** [comment]
That is something that I do want to add for better accessibility! This will eventually come in a future release

**Comment by CapnHanSolo** [comment]
+1 for font size.

For me the compact view is not compact enough due to font size. Also I've grown used to smaller comment fonts and would like the option very much.

**Comment by KarkanAlzwayed** [comment]
Just wanted to chime in on this. This is literally the only reason why I don't use this app exclusively. I really love this app, but the small font is killing it for me :(

**Comment by hjiangsu** [comment]
Hey! Before I get started on this, I wanted to ask a few questions to the community.

For this, how would you best like it to be?
- Have 2 options for font customization - one for posts and one for comments
  - If this is the preferred way, should the font size affect all text within the posts/comments? Or just specific parts (e.g., title, body, etc)?
- Have a global font size adjustment, where you have a single slider to adjust your preferred font size. This would equally increase the font sizes for all text within the app

**Comment by KarkanAlzwayed** [comment]
I personally would do everything separate just like how infinity for reddit does it. I'd create a section inside of the "theming" that has font size and font family (global), then title with font size and font family, then content/comments with font size and font family. If that's too much work, then leave the font family out, it's just very nice to be able to choose what font to use.

**Comment by hjiangsu** [comment]
Could you show me a screenshot of what you mean to allow me to visualize it a bit better? Thanks!

**Comment by KarkanAlzwayed** [comment]
Absolutely, 
![Screenshot_20230703_201753_Infinity](https://github.com/hjiangsu/thunder/assets/49320606/6fed6e61-db60-4da1-8ea6-545d532124e1)
![Screenshot_20230703_201804_Infinity](https://github.com/hjiangsu/thunder/assets/49320606/7eb19dd6-9957-4699-9063-1b3f7f0103b6)
![Screenshot_20230703_201824_Infinity](https://github.com/hjiangsu/thunder/assets/49320606/d6dce00a-749a-462a-863c-21c9d9c225a5)

**Comment by hjiangsu** [comment]
Just an update, there will be an initial implementation for this in the next nightly release. I would like to get some feedback for the scale of things (whether they should be smaller/larger) and also whether theres any other parts that should be adjusted as well!

Keep in mind this is very rough at the moment, and it'll slowly get better as I get more feedback!

Some initial screenshots:
![simulator_screenshot_BC420F3A-69D7-4A27-ABD9-74D29F99EA41](https://github.com/hjiangsu/thunder/assets/30667958/c2f67f90-a48b-478d-81e4-b7215df95b63)

![simulator_screenshot_3A86E24E-5AC9-41E3-A7A2-80DED3499131](https://github.com/hjiangsu/thunder/assets/30667958/9a5cfcc4-b9e5-415c-88a3-d79cb5160545)

![simulator_screenshot_F9226954-2134-411D-8BCC-53F3FE221D0E](https://github.com/hjiangsu/thunder/assets/30667958/9efd6883-1e29-4118-8549-2f8f0d5c4096)

**Comment by hjiangsu** [comment]
I have just released the pre-release version v0.2.1-alpha.1+11 which has the initial implementation for font scaling - its still rough but let me know your initial thoughts on it!

**Comment by KarkanAlzwayed** [comment]
That looks very nice. I have both set to "extra large" so I can see better. Lol
If you need more testing please let us know.

**Comment by KarkanAlzwayed** [comment]
Also, is a slider with digits possible? I could still go for a bigger font.

**Comment by machinaeZER0** [comment]
A slider with digits would be cool, though the dropdown is quite clean and easy to use (for some reason a lot of the other lemmy clients I tried with font sliders were weirdly hard to control?)

I'd mentioned this in the nightly announcement post, but at least on my phone this latest update made the default slightly smaller for me:

Old:
![Screenshot_20230704-082754](https://github.com/hjiangsu/thunder/assets/3624023/968ff1cf-d3ea-4318-b45d-a0f98c98f525)

Latest Build:
![Screenshot_20230704-083428](https://github.com/hjiangsu/thunder/assets/3624023/7d13dae2-a373-49f4-824b-6889d3408251)

Not sure if the new base size is just more true to my system font size, but in practice the change has made it closer to the Small font than the Large font, which is a bit less comfortable to read :( the new Large size is much bigger than I would probably use, personally. Would it be possible to change the base sizes back to what they were before?

**Comment by KarkanAlzwayed** [comment]
> A slider with digits would be cool, though the dropdown is quite clean and easy to use (for some reason a lot of the other lemmy clients I tried with font sliders were weirdly hard to control?)
> 
> I'd mentioned this in the nightly announcement post, but at least on my phone this latest update made the default slightly smaller for me:
> 
> Old: ![Screenshot_20230704-082754](https://user-images.githubusercontent.com/3624023/251015297-968ff1cf-d3ea-4318-b45d-a0f98c98f525.png)
> 
> Latest Build: ![Screenshot_20230704-083428](https://user-images.githubusercontent.com/3624023/251015319-7d13dae2-a373-49f4-824b-6889d3408251.png)
> 
> Not sure if the new base size is just more true to my system font size, but in practice the change has made it closer to the Small font than the Large font, which is a bit less comfortable to read :( the new Large size is much bigger than I would probably use, personally. Would it be possible to change the base sizes back to what they were before?

I've experienced the same issue. I don't remember what I did to work around it. I had to set one of them to regular size and the other to largest. It was weird that setting up the comments to largest and base to largest makes the comments too small. Also, the new alpha release broke the commenting function for me. Every time I comment, it exits the whole thread when I submit my comment.

**Comment by hjiangsu** [comment]
> Not sure if the new base size is just more true to my system font size, but in practice the change has made it closer to the Small font than the Large font, which is a bit less comfortable to read :( the new Large size is much bigger than I would probably use, personally. Would it be possible to change the base sizes back to what they were before?

I'm still working on adjusting the font scales to be better, but I think that'll come with time and more feedback! I can increase the base font size a bit and see if thats a bit better, and also maybe add in a couple more options to choose from to be more granular

**Comment by hjiangsu** [comment]
> I've experienced the same issue. I don't remember what I did to work around it. I had to set one of them to regular size and the other to largest. It was weird that setting up the comments to largest and base to largest makes the comments too small. 

Yeah, this is still very much open to be tweaked!

> Also, the new alpha release broke the commenting function for me. Every time I comment, it exits the whole thread when I submit my comment.

The latest build is a pre-release version, so not necessarily an alpha build per-say, so there are a few things that are broken.  The bug you mentioned should be fixed in the upcoming nightly build!

**Comment by machinaeZER0** [comment]
> I can increase the base font size a bit and see if thats a bit better

That would be great if you don't mind - sorry, feel like such a stickler 😆 One of the first things I noticed about Thunder was that the font size felt just right for me, hahah.

**Comment by hjiangsu** [comment]
No worries! I also wanted to keep the general font size the same on the default base setting but there seems to be something at play here.

When I set the font scale to 1.0, which should be the base scale, it makes the text on my iPhone much much larger than before. It might have a different behaviour on Android though. That's why I reduced the font scale to 0.9 which matched closer on my iPhone to what it was originally. However in your case, it seemed to have made it smaller than before 😅

**Comment by machinaeZER0** [comment]
Oh... That's really odd! I wonder why iOS would interpret it that way... I
don't know much about coding unfortunately, but if this was specific to the
size of post titles, could it have something to do with them being
classified as header text or something? And if so, maybe iOS is handling
that text like a header, which my default might display as larger than
normal text?

Total shot in the dark, so probably wrong, hahah. Where in the code does
font stuff live (perhaps feed specific)? I definitely can't fix anything
but I was interested in feed code anyway and maybe having it in front of me
would glean some insights?

**Comment by machinaeZER0** [comment]
(by default, not my default)

**Comment by hjiangsu** [comment]
Here is the scaling for the different font sizes at the moment:
- I've tweaked to base font size to be 0.95, this should hopefully improve things on both fronts (make text smaller on iOS, make text larger on Android
- Eventually, I'll probably make the scaling specific to each OS (different for Android/iOS) to fix these issues

This is where I have the font scales:

https://github.com/hjiangsu/thunder/blob/6a46c0620e20ecd31eb50700ec04c514aefd69de/lib/core/enums/font_scale.dart#L9-L20

**Comment by machinaeZER0** [comment]
Sounds good, thank you!

**Comment by hjiangsu** [comment]
Closing as its been released in v0.2.1+11!
